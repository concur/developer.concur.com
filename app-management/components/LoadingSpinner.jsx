import React, { PropTypes } from 'react';

const LoadingSpinner = ({ loading, message }) => (
  loading ? (
    <h3>
      <i className="fa fa-spinner fa-spin" />
      &nbsp;
      {message || 'Loading...'}
    </h3>
  ) : null
);

LoadingSpinner.propTypes = {
  loading: PropTypes.bool,
  message: PropTypes.string,
};

export default LoadingSpinner;
