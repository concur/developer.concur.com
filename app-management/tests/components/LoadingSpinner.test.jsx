import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import LoadingSpinner from '../../components/LoadingSpinner';

describe('<LoadingSpinner />', () => {
  it('should render when loading and given a message', () => {
    const message = 'Fetching data';
    const spinner = shallow(<LoadingSpinner loading={true} message={message} />);

    expect(shallowToJson(spinner)).toMatchSnapshot();
  });

  it('should display a default message if no message is given', () => {
    const spinner = shallow(<LoadingSpinner loading={true} />);

    expect(shallowToJson(spinner)).toMatchSnapshot();
  });

  it('should display nothing if not loading', () => {
    const spinner = shallow(<LoadingSpinner loading={false} />);

    expect(shallowToJson(spinner)).toMatchSnapshot();
  });
});
