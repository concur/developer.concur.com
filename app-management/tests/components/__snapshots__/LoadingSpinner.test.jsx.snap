exports[`<LoadingSpinner /> should display a default message if no message is given 1`] = `
<h3>
  <i
    className="fa fa-spinner fa-spin" />
   
  Loading...
</h3>
`;

exports[`<LoadingSpinner /> should display nothing if not loading 1`] = `null`;

exports[`<LoadingSpinner /> should render when loading and given a message 1`] = `
<h3>
  <i
    className="fa fa-spinner fa-spin" />
   
  Fetching data
</h3>
`;
